python/src/UserDefinedMetropolisHastings_doc.i.in

```diff
@@ -0,0 +1,85 @@
+%feature("docstring") OT::UserDefinedMetropolisHastings
+"User-defined Metropolis-Hastings method.
+
+Refer to :ref:`bayesian_calibration`, :ref:`metropolis_hastings`.
+
+Available constructor:
+    UserDefinedMetropolisHastings(*targetDistribution, initialState, proposal, linkFunction, marginalIndices*)
+
+    UserDefinedMetropolisHastings(*targetLogPDF, support, initialState, proposal, linkFunction, marginalIndices*)
+
+Parameters
+----------
+targetDistribution : :class:`~openturns.Distribution`
+    Target distribution sampled
+targetLogPDF : :class:`~openturns.Function`
+    Target log-density up to an additive constant
+support : :class:`~openturns.Domain`
+    Support of the target when defined with *targetLogPDF*
+initialState : sequence of float
+    Initial state of the chain
+proposal : :class:`~openturns.Distribution`
+    Proposal distribution, its parameters depend on the state of the chain.
+linkFunction : :class:`~openturns.Function`
+    Link between the state of the chain and the parameters of *proposal*.
+marginalIndices : sequence of int, optional
+    Indices of the components to be updated.
+    If not specified, all components are updated.
+    The number of updated components must be equal to the dimension of *proposal*.
+
+Examples
+--------
+>>> import openturns as ot
+>>> import math as m
+>>> ot.RandomGenerator.SetSeed(0)
+
+Sample from a target distribution defined through its log-PDF (defined up to some additive constant) and its support:
+
+>>> log_density = ot.SymbolicFunction('x', 'log(2 + sin(x)^2)')
+>>> support = ot.Interval([0.0], [2.0 * m.pi])
+
+Apply a *Metropolis adjusted Langevin algorithm* (MALA) [robert2015]_ (page 10).
+The idea is to use a normal proposal distribution,
+whose mean will depend on the state of the chain
+(but will not be equal to that state, otherwise the algorithm
+would be easier to implement with the :class:`~openturns.RandomWalkMetropolisHastings` class).
+
```

Comment:
Maybe separate explanations of the general principle of the algo in the top or into a Notes section on the (and the difference to RWMH like you do here) instead of the inside the example.

Reply:
I have made an addition to the API page in order to better describe what the class does. Would the following algorithmic description be adequate? It draws on the theory (and notations) established in [The Metropolis-Hastings Algorithm](https://openturns.github.io/openturns/1.19/theory/data_analysis/metropolis_hastings.html ) theory page.

![image](https://user-images.githubusercontent.com/31989332/183707061-c9cefb97-1d5d-4754-9e88-56de6d996d5f.png)

Reply:
Marking this resolved, but feel free to ask for additional explanations or modifications if needed.